Tutorials/Add-Google-Authentication/index.md

```diff
@@ -1,134 +1,274 @@
 
-This will update the package and dependencies, and clear out the issues.
+5. [Optional] In the **Test Users** tab, add the test users that can access the application. Click **Save and Continue**.
 
-Now we need to build the website again then try to run it. Hopefully you reach the Umbraco page like I did:
+6. In the **Summary** tab, verify the details provided. Click **Back to Dashboard** or **Submit for verification**.
 
-![Umbraco homepage](images/umbracoHomepage.png)
+7. Click on **Credentials** from the left-side navigation menu. Click on **Create Credentials** and select **OAuth Client ID**.
+    ![OAuth Client Id](images/OAuth_Client_Id.png)
 
-Next, we will install the package we need for Google authentication, for that paste the following in the NuGet Package console:
+8. Select **Web Application** from the **Application type** drop-down.
 
-`Install-Package UmbracoCms.IdentityExtensions.Google`
+9. Enter the application **Name**, **Authorized JavaScript origins**, **Authorized redirect URIs** and click **Create**.
+    ![Credentials Details](images/Credentials_v9.png)
 
-Like before, wait for the package to install then build and run the website.
+A popup appears displaying the **ClientId** and **ClientSecret**. You will need these values later while configuring your solution. 
 
-Now you have installed both packages, and your site still works (hopefully)! Now we have to configure the files to allow Google Authentication as login.
+:::note
+The **ClientId** and **ClientSecret** can always be accessed from the **Credentials** tab in **APIs & Services** menu.
+:::
 
-## Configuring the solution to allow Google logins
-
-The first thing to do is locate the files we installed from the packages. They are located in the App_Code/App_start folder. There are two files we need to edit, first one is the file called UmbracoGoogleAuthExtensions.cs. The only thing you need to touch here is the callback path, this is what we set on the API as the redirect URL. If you followed my example it should be set to /google-signin:
-
-![Callback path](images/callbackPath.png)
-
-The second file we need to configure is the UmbracoStandardOwinStartup.cs file. Here we need to add the following code:
-
-```csharp
-var clientId = ConfigurationManager.AppSettings["GoogleOAuthClientID"];
-var secret = ConfigurationManager.AppSettings["GoogleOAuthSecret"];
-app.ConfigureBackOfficeGoogleAuth(clientId, secret);
-```
+## Integrating Google Auth in Visual Studio
 
-Add this in the file as seen here:
+Now that you have the Google API set up, open your existing solution in Visual Studio. If you don't know how to clone down a Cloud site, see the [Working with Visual Studio](../../Umbraco-Cloud/Set-Up/Working-with-Visual-Studio/) article.
 
-![Set client info](images/setClientInfo.png)
+### Installing a Nuget Package
 
-The ConfigurationManager is missing a dependency so add the following at the top: `using System.Configuration;`
+*Option 1*
```

Comment:
When we have two options for accomplishing the same thing, would it make sense to explain when to use which? And perhaps explain the difference between the two?

Reply:
Done.. I have added some info around both the options and added a reference link to the Microsoft docs 🙂

Reply:
Looks great! Thanks 🙌